pkg/3scale/amp/component/backend.go

```diff
@@ -398,9 +403,15 @@ func (backend *Backend) buildBackendWorkerEnv() []v1.EnvVar {
 	result = append(result,
 		helper.EnvVarFromSecret("CONFIG_EVENTS_HOOK", "system-events-hook", "URL"),
 		helper.EnvVarFromSecret("CONFIG_EVENTS_HOOK_SHARED_SECRET", "system-events-hook", "PASSWORD"),
-		v1.EnvVar{Name: "CONFIG_WORKER_PROMETHEUS_METRICS_PORT", Value: BackendWorkerMetricsPortStr},
-		v1.EnvVar{Name: "CONFIG_WORKER_PROMETHEUS_METRICS_ENABLED", Value: "true"},
 	)
+
+	if backend.Options.WorkerMetrics != nil {
```

Comment:
Shouldn't we be checkind that is not null AND that is set to true? :thinking:

Reply:
That field type is `*bool`. But it is not intended to hold true or false. The real information is on pointer existence. As it is not reconciliable. It is enabled for operator based deployment and disabled for template based deployment.

Changing it to `bool`. It is confusing.

Reply:
:+1: